docs/proposals/001-k8s-status.md

```diff
@@ -0,0 +1,432 @@
+  message: CoreDNS deployed
+```
+
+## User facing changes
+<!--
+This section MUST describe any user-facing changes that this feature brings, if
+any. If an API change is required, the affected endpoints MUST be mentioned. If
+the output of any k8s command changes, the difference MUST be mentioned, with a
+clear example of "before" and "after".
+-->
+
+### `k8s status` command
+
+The output format of `k8s status` will change.
+
+#### Old output
+
+```bash
+$ k8s status
+status: ready
+high-availability: no
+datastore:
+  type: k8s-dqlite
+  voter-nodes:
+    - 10.87.24.119:6400
+  standby-nodes: none
+  spare-nodes: none
+network:
+  enabled: true
+dns:
+  enabled: true
+  cluster-domain: cluster.local
+  service-ip: 10.152.183.104
+  upstream-nameservers:
+  - /etc/resolv.conf
+ingress:
+  enabled: false
+  default-tls-secret: ""
+  enable-proxy-protocol: false
+load-balancer:
+  enabled: false
+  cidrs: []
+  l2-mode: false
+  l2-interfaces: []
+  bgp-mode: false
+  bgp-local-asn: 0
+  bgp-peer-address: ""
+  bgp-peer-asn: 0
+  bgp-peer-port: 0
+local-storage:
+  enabled: true
+  local-path: /var/snap/k8s/common/rawfile-storage
+  reclaim-policy: Delete
+  default: true
+gateway:
+  enabled: true
+```
+
+#### New output
+
+With `--output-format=plain`, a simple overview is shown, with `yaml` or `json`
+more fields are shown:
+
+```bash
+$ k8s status
+status: ready
+high-availability: no
+datastore:
+  type: k8s-dqlite
+  voter-nodes:
+    - 10.87.24.119:6400
+  standby-nodes: none
+  spare-nodes: none
+network:
+  enabled: true
+  message: Cilium is deployed
+dns:
+  enabled: true
+  message: CoreDNS is deployed
+ingress:
+  enabled: false
+  message: ""
+load-balancer:
+  enabled: true
+  message: Cilium configured for LoadBalancer services
+local-storage:
+  enabled: true
+  message: Rawfile LocalPV is deployed
+gateway:
+  enabled: false
+  message: Cilium GatewayAPI support could not be enabled, error was "..."
+```
+
+When the output format is `json` or `yaml`, more fields may be shown:
+
+```bash
+$ k8s status --output-format=yaml
+...
+network:
+  enabled: true
+  message: Cilium is deployed
+  timestamp: 2020-01-01 10:00:00
+  version: v1.15.2
+...
+```
+
+## Alternative solutions
+<!--
+This section SHOULD list any possible alternative solutions that have been or
+should be considered. If required, add more details about why these alternative
+solutions were discarded.
+-->
+
+none
+
+## Out of scope
+<!--
+This section MUST reference any work that is out of scope for this proposal.
+Out of scope items are typically unknowns that we do not yet have a clear idea
+of how to solve, so we explicitly do not tackle them until we have more
+information.
+
+This section is very useful to help guide the implementation details section
+below, or serve as reference for future proposals.
+-->
+
+### `message` field
+
+The `message` field is the only information that will be presented to the user.
```

Comment:
I'm +1 on this for errors but I don't think it provides value if we would e.g. add an additional "Cilium deployed" message if all is good.